File: leetcode/easy/javascript/Valid Parentheses.js
var isValid = function(s) {
    const stack = [] //push,pop用空配列用意

    for(let i=0;i<s.length;i++) { //引数の長さだけループ
        let str = s.charAt(i) //charAtメソッドで指定位置の文字を取ってくる
        switch(str) {
            case "(":stack.push(")") //対応文字の一つ目だったら、２つ目を配列にプッシュ
                break;
            case "[":stack.push("]") //対応文字の一つ目だったら、２つ目を配列にプッシュ
                break;
            case "{":stack.push("}") //対応文字の一つ目だったら、２つ目を配列にプッシュ
                break;
            default:
                if(str!==stack.pop()) { //プッシュされていた文字と一致しない場合falseを返す
                    return false
                }
        }
    }
    return stack.length==0 //ifの条件に引っかからず全て取り出せた場合、全一致としてtrueを返す
};

File: leetcode/easy/python/Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        opcl = dict(('()', '[]', '{}')) #dictで辞書作成
        stack = []
        for idx in s:
            if idx in '([{': #値が([{いずれかの場合
                stack.append(idx) #配列に入れる
            elif len(stack) == 0 or idx != opcl[stack.pop()]: #配列がループ途中で０のまま、またはループ時の値と辞書の値が一致しない場合
                return False
        return len(stack) == 0
